src/pages/_app.tsx

```diff
@@ -24,7 +25,9 @@ const BeaconRegistrationApp: FunctionComponent<AppProps> = ({
         <GovUKMetadata />
       </Head>
 
-      <Component {...pageProps} />
+      <AuthProvider session={pageProps.session}>
+        <Component {...pageProps} />
+      </AuthProvider>
```

Comment:
Provider for exposing session information, [see docs](https://next-auth.js.org/getting-started/client#provider)

Reply:
Noice 👌. Does the `<AuthProvider>` work OK with SSR?  Had real headaches playing around with React Context and NextJS SSR earlier in the project when trying to do form validation.

Reply:
I'll check now.  There is a `getSession` function that can be used in
getServerSideProps, see docs:
https://next-auth.js.org/getting-started/client#getsession

And here:
https://next-auth.js.org/tutorials/securing-pages-and-api-routes#server-side

Reply:
I would suggest we wrap it in some kind of AuthInterface that has a gateway implementing to encapsulate any changes that are made